src/page/page_cache.rs

```diff
@@ -386,12 +392,6 @@ impl<PM, P, R> PageCache<PM, P, R>
             .rev()
             .collect();
 
-        // Short circuit merging and fix-up if we only
-        // have one frag.
-        if combined.len() == 1 {
-            return Some((combined[0].clone(), head.into()));
-        }
-
```

Comment:
this could return a fragment that was paged-out and just read off the log before we have a chance to store it